actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,11 +92,20 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
+      # This is in the case that the browser client wants
+      # to open the connection regardless of whether a
+      # protocol was negotiated successfully.
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        @close()
+      else
+        @disconnected = false
+        @subscriptions.reload()
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        ActionCable.log("WebSocket is closing because there is no common protocol. Stopping ConnectionMonitor.")
+        @monitor.stop()        
```

Comment:
This will prevent the connection monitor from infinitely attempting to reconnect (as noted in https://github.com/rails/rails/pull/23976#issuecomment-197627120), but leaves you disconnected until you reload your browser. Think we should notify subscriptions of this state so they can manage / instruct reloading.